lib/Cryptol.cry

```diff
@@ -323,3 +324,133 @@ groupBy : {each,parts,elem} (fin each) =>
   [parts * each] elem -> [parts][each]elem
 groupBy = split`{parts=parts}
 
+/**
+ * Logical implication
+ */
+(==>) : {a} a -> a -> a
+a ==> b = not a || b
```

Comment:
Done.

On Thu, Dec 31, 2015 at 9:10 AM, brianhuffman <notifications@github.com>
wrote:

> In lib/Cryptol.cry
> <https://github.com/GaloisInc/cryptol/pull/299#discussion_r48663604>:
>
> > @@ -323,3 +324,133 @@ groupBy : {each,parts,elem} (fin each) =>
> >    [parts * each] elem -> [parts][each]elem
> >  groupBy = split`{parts=parts}
> >
> > +/**
> > + * Logical implication
> > + */
> > +(==>) : {a} a -> a -> a
> > +a ==> b = not a || b
>
> Weren't we going to specialize this to type Bit? I would actually prefer
> to redefine it as
>
> (==>) : Bit -> Bit -> Bit
> a ==> b = if a then b else True
>
> to make it explicitly lazy in the second argument. (According to the
> Cryptol semantics that hasn't been written yet, I think bitwise operators
> like (||) will be strict in both arguments.) This will be important for
> using (==>) to state properties with preconditions that guard against
> undefined behavior.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GaloisInc/cryptol/pull/299/files#r48663604>.
>